src/query/print.go

```diff
@@ -182,20 +182,37 @@ func (p *printer) PrintFields(fields []string) bool {
 // This isn't as simple as using reflect.Value.FieldByName since the print names
 // are different to the actual struct names.
 func (p *printer) findField(field string) reflect.StructField {
-	t := reflect.ValueOf(p.target).Elem().Type()
-	for i := 0; i < t.NumField(); i++ {
-		if f := t.Field(i); p.fieldName(f) == field {
-			return f
+	// There isn't a 1-1 mapping between the field and its structure. Internally, we use
+	// things like named vs unnamed structures which reflect the same field from the user
+	// perspective. The function below takes that into consideration.
+	findFieldStruct := func(value interface{}, fieldName string) (reflect.StructField, bool) {
+		v := reflect.ValueOf(value).Elem()
+		t := v.Type()
+
+		resIndex := -1
+		for i := 0; i < v.NumField(); i++ {
+			if f := t.Field(i); p.fieldName(f) == fieldName {
+				if resIndex == -1 {
+					resIndex = i
```

Comment:
I wonder if we can return early here if the field value isn't zero and skip the check on the next line?

```
if !v.Field(i).IsZero() {
    return t.Field(i), true
} else if resindex == -1 {
    resIndex = i 
}
```

Reply:
Ya, sounds better